File: Python/Binary Search Tree/700. Search in a Binary Search Tree.py
""" 700. Search in a Binary Search Tree
Difficulty: Easy
Submission Link: https://leetcode.com/problems/search-in-a-binary-search-tree/submissions/1381342036/ """

class Solution(object):
    def searchBST(self, root, val):
        cur = root
        if cur is None:
            return None
        if val == cur.val:
            return cur
        if val > cur.val:
            return self.searchBST(cur.right, val)
        else:
            return self.searchBST(cur.left, val)

File: JavaScript/Binary Search Tree/700. Search in a Binary Search Tree.js
/* 700. Search in a Binary Search Tree
Difficulty: Easy
Submission Link: https://leetcode.com/problems/search-in-a-binary-search-tree/submissions/1381341594/ */

var searchBST = function (root, val) {

    let current = root;
    while (current) {
        if (val === current.val) {
            return current;
        } else if (val < current?.val) {
            current = current.left;
        } else {
            current = current.right;
        }
    }
    
    return null;
};